File: app/src/main/java/com/chooloo/www/callmanager/ui/about/AboutContract.java
package com.chooloo.www.callmanager.ui.about;

import com.chooloo.www.callmanager.ui.base.BaseContract;


public class AboutContract implements BaseContract {

    interface View extends BaseContract.View {

        void setUp();

        void openChangelog();

        void openSource();

        void sendEmail();

        void reportBug();

        void rateApp();

        void donate();
    }

    interface Presenter<V extends View> extends BaseContract.Presenter<V> {

        void onChangelogClick();

        void onOpenSourceClick();

        void onSendEmailClick();

        void onReportBugClick();

        void onRateAppClick();

        void onDonateClick();
    }
}

File: app/src/main/java/com/chooloo/www/koler/ui/about/AboutContract.kt
package com.chooloo.www.koler.ui.about

import com.chooloo.www.koler.ui.base.BaseContract

interface AboutContract : BaseContract {
    interface View : BaseContract.View {
        fun openSource()
        fun sendEmail()
        fun reportBug()
        fun rateApp()
        fun donate()
    }

    interface Presenter<V : View> : BaseContract.Presenter<V> {
        fun onOpenSourceClick()
        fun onSendEmailClick()
        fun onReportBugClick()
        fun onRateAppClick()
        fun onDonateClick()
    }
}